e2e/scenario_helpers_test.go

```diff
@@ -15,16 +17,52 @@ import (
 	"github.com/tidwall/gjson"
 )
 
+// it's important to share context between tests to allow graceful shutdown
+// cancellation signal can be sent before a test starts, without shared context such test will miss the signal
+var testCtx = setupSignalHandler()
+
+// setupSignalHandler handles OS signals to gracefully shutdown the test suite
+func setupSignalHandler() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+	ch := make(chan os.Signal, 2)
+	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		// block until signal is received
+		<-ch
+		fmt.Println("Received cancellation signal, gracefully shutting down the test suite. Cancel again to force exit.")
+		cancel()
+
+		// block until second signal is received
```

Comment:
I'm probably missing something, though wouldn't we leak this goroutine in cases where the user does a graceful shutdown?

Reply:
This function is called only once on a startup.
And after every test shutdown the test application is closed.